parse_test.go

```diff
@@ -428,3 +429,30 @@ func TestParse_stdfs(t *testing.T) {
 		t.Errorf("foo: want %q, have %q", want, have)
 	}
 }
+
+func TestParse_shortSameCase(t *testing.T) {
+	t.Parallel()
+
+	newFS := func() *ff.FlagSet {
+		fs := ff.NewFlagSet(t.Name())
+		_ = fs.String('a', "apple", "", "foo string")
+		_ = fs.String('A', "apricot", "", "FOO string")
+		return fs
+	}
+	args := []string{"-a", "-A"}
+
+	t.Run("no env", func(t *testing.T) {
+		t.Parallel()
+		if err := ff.Parse(newFS(), args); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("WithEnv", func(t *testing.T) {
+		t.Parallel()
+		if err := ff.Parse(
+			newFS(), args, ff.WithEnvVars(),
+		); err == nil || !errors.Is(err, ff.ErrDuplicateFlag) {
```

Comment:
Isn't this enough?

```suggestion
		); !errors.Is(err, ff.ErrDuplicateFlag) {
```

I'm just curious, I'm fine with the test you have, it's clearer